OSDOCS-13509: add assembly, topic map, and concept (#92976)
openshift/openshift-docs: modules/nw-ingress-gateway-api-implementation.adoc

// Modules included in the following assemblies:
//
// * networking/gateway-api.adoc

:_mod-docs-content-type: CONCEPT
[id="nw-ingress-gateway-api-implementation_{context}"]

> = {product-title}'s Gateway API implementation

Author: its a good question, i wonder that myself. @ahardin-rh what do you think?

> = {product-title}'s Gateway API implementation

Author: i remember @Miciah mentioned [here](https://github.com/openshift/openshift-docs/pull/89357?notification_referrer_id=NT_kwDOAOLCOLQxNTA0ODA5NTY5MzoxNDg2MDg1Ng#discussion_r1985549402) that maybe we don't want to do that. has this changed?

The Ingress Operator manages the lifecycle of Gateway API CRDs in a way that enables other vendor implementations to make use of CRDs defined in an {product-title} cluster.

> In some situations, the Gateway API provides one or more fields that an given vendor implementation does not support, but that implementation is otherwise compatible in schema with the rest of the fields. These "dead fields" can result in disruption of Ingress workloads, improperly provisioned applications and services, and security related issues. Because {product-title} uses a specific version of Gateway API CRDs, any use of third-party implementations of Gateway API must conform to the {product-title} implementation to ensure that all fields work as expected.

Reviewer: `In some situations, the Gateway API provides one or more fields that an given vendor implementation `
is the 'an' supposed to be 'any' or 'a'

Author: "a" for sure. this is just a leftover issue, the sentence was previously `.. fields that an implementation`

Any CRDs created within an {product-title} {product-version} cluster will be compatibly versioned and maintained by the Ingress Operator. In the case that CRDs are already present but were not previously managed by the Ingress Operator, the Ingress Operator will check if these configurations are compatible with the Gateway API version supported by {product-title}, and create an admin-gate that requires your acknowledgment of CRD succession.

[IMPORTANT]
====

> If you are updating your cluster from a previous {product-title} version that contains Gateway API CRDs, you must make changes to these resources such that it exactly matches the version supported by {product-title}. Otherwise, you will not be able to update your cluster. This is because those CRDs were not managed by {product-title}, and could contain functionality that is unsupported by Red{nbsp}Hat.

Reviewer 2: ```suggestion
If you are updating your cluster from a previous {product-title} version that contains Gateway API CRDs, you must change those resources so that they exactly match the version supported by {product-title}. Otherwise, you cannot update your cluster because those CRDs were not managed by {product-title} and could contain functionality that is unsupported by Red{nbsp}Hat.
```
